src/darwin/Framework/CHIP/ServerEndpoint/MTRServerCluster.h

```diff
@@ -0,0 +1,111 @@
+/**
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import <Matter/MTRAccessGrant.h>
+#import <Matter/MTRDefines.h>
+#import <Matter/MTRServerAttribute.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * A representation of a server cluster implemented by an MTRDeviceController.
+ */
+MTR_NEWLY_AVAILABLE
+@interface MTRServerCluster : NSObject
+
+- (instancetype)init NS_UNAVAILABLE;
++ (instancetype)new NS_UNAVAILABLE;
+
+/**
+ * The provided clusterID must not be MTRClusterIDTypeDescriptorID; see
+ * initDescriptorCluster.
+ *
+ * Otherwise, it must be a valid cluster identifier.  That means:
+ *
+ * * In the range 0-0x7FFF for standard clusters.
+ * * In the range 0xVVVVFC00-0xVVVVFFFE for vendor-specific clusters, where VVVV
+ *   is the vendor identifier.
+ *
+ * The provided clusterRevision must be in the range 1-65535.
+ *
+ */
+- (nullable instancetype)initWithClusterID:(NSNumber *)clusterID clusterRevision:(NSNumber *)clusterRevision;
```

Comment:
Nit: could just be `initWithClusterID:revision:`. Also should revision be `NSInteger` instead of `NSNumber`?

Reply:
I was told by @woody-apple and @anush-apple to stick to NSNumber in APIs in general, back when we were doing our initial API reviews on this stuff.  But we can do whatever; what's the usual convention?

Reply:
The type is tracked in https://github.com/project-chip/connectedhomeip/issues/31797

I changed the selector as suggested.